### ISSUE TITLE
Change C++ logging library to `spdlog` or `glog`

### ISSUE THREAD

**Issue by garth-wells** [FEniCS/dolfinx#3195]
Some time ago we adopted [loguru](https://github.com/emilk/loguru) for C++ logging. At the time we considered several options (see https://github.com/FEniCS/dolfinx/issues/3). The history is a bit hard to follow, but we did have a PR for using [spdlog](https://github.com/gabime/spdlog) (https://github.com/FEniCS/dolfinx/pull/311), but eventually adopted loguru (https://github.com/FEniCS/dolfinx/pull/417).

A consideration at the time was a reluctance to introduce a substantial dependency that was not widely packaged. loguru is light and we included the source in the DOFLINx tree.

In the meantime, 
- loguru is not actively maintained. The code has many pragmas to ignore compiler warnings, we're having to edit the code to add more as compilers add stricter tests.
- spdlog and [glog](https://github.com/google/glog) are actively maintained and widely used.
- spdlog and glog are available in conda, brew, Debian/Ubuntu, Spack, vcpkg , and other package managers

Should we switch from loguru to spdlog or glog? 

@jhale, @chrisrichardson

**Comment by chrisrichardson** [comment]
glog is also worth considering. It would be less disruptive as the syntax is the same as what we have in loguru.
loguru works well, so I'd like to confirm that the same or better features exist before switching.

**Comment by jhale** [comment]
I support the change but am also interested in glog.

Both glog and spdlog are available in vcpkg, homebrew, Debian, RedHat, conda (feedstock) and Spack so no issues there.

glog (unlike spdlog) has a Windows CI. It is also a drop in replacement for loguru.

We are not logging heavily in tight loops so performance differences are not hugely important for us.

@chrisrichardson and I have nearly finished the Windows build and have identified loguru as producing strange linking (!) errors when compiling the Python headers, so we're keen to make a switch to something more modern.

**Comment by garth-wells** [comment]
I've updated the title to add glog to spdlog.

**Comment by chrisrichardson** [comment]
https://github.com/FEniCS/dolfinx/tree/chris/glog - seems an easy drop-in.

**Comment by chrisrichardson** [comment]
On further testing, I think spdlog is a better choice:
https://github.com/FEniCS/dolfinx/compare/chris/spdlog
